### PR TITLE
Update detray to v45

### PR DESCRIPTION
**Pull request by beomki-yeo**
Update detray to v45

### CODE REVIEW

File: examples/simulation/simulate_telescope.cpp [hunk 1]
```diff
@@ -79,11 +81,17 @@ int simulate(std::string output_directory, unsigned int events,
...
 
     // Origin of particles
-    auto generator =
+    using generator_type =
         detray::random_track_generator<traccc::free_track_parameters,
-                                       uniform_gen_t>(
-            pg_opts.gen_nparticles, pg_opts.vertex, pg_opts.vertex_stddev,
-            pg_opts.mom_range, pg_opts.theta_range, pg_opts.phi_range);
+                                       uniform_gen_t>;
+    generator_type::configuration gen_cfg{};
```
Comment (niermann999) [inline examples/simulation/simulate_telescope.cpp:87]: Have you considered putting the `generator_type::configuration` as a member into the `particle_gen_options` ?

Comment (beomki-yeo) [reply examples/simulation/simulate_telescope.cpp:87]: Let's just go like this

File: device/common/include/traccc/finding/device/propagate_to_next_surface.hpp [hunk 2]
```diff
@@ -13,6 +13,9 @@
 #include "traccc/edm/measurement.hpp"
 #include "traccc/edm/track_parameters.hpp"
 
+// Covfie include(s).
+#include <covfie/core/field.hpp>
```
[the remark this answers is not in the document]
Comment (beomki-yeo) [reply device/common/include/traccc/finding/device/propagate_to_next_surface.hpp:17]: No. Removed

File: core/include/traccc/finding/finding_algorithm.hpp [hunk 3]
```diff
@@ -39,6 +39,7 @@ template <typename stepper_t, typename navigator_t>
 class finding_algorithm
     : public algorithm<track_candidate_container_types::host(
           const typename navigator_t::detector_type&,
+          const typename stepper_t::magnetic_field_type&,
```
[the remark this answers is not in the document]
Comment (beomki-yeo) [reply core/include/traccc/finding/finding_algorithm.hpp:42]: View type should be a reference always?
I should pass it as a value in the device code, but reference works fine in host side as unit test suggests.

[the remark this answers is not in the document]
Comment (beomki-yeo) [reply core/include/traccc/finding/finding_algorithm.hpp:42]: I just fixed the member of kalman fitter to be a value. but didn't fix others

**Review by niermann999 — APPROVED** [review]

**Pull request merged**